store-and-retrieve-data/use-storagehub-sdk/create-a-bucket.md

```diff
@@ -74,120 +39,155 @@ async function run() {
+    ```
 
-```bash
-ts-node index.ts
-```
+??? code "View complete `mspService.ts` file"
 
-The response should include something like this:
+    ```ts title="mspService.ts"
+    --8<-- 'code/store-and-retrieve-data/use-storagehub-sdk/create-a-bucket/msp-service-with-value-props.ts'
+    ```
 
---8<-- 'code/store-and-retrieve-data/use-storagehub-sdk/create-a-bucket/output-02.html'
+### Add Method to Create a Bucket
 
-## Check If Derived Bucket ID Isn’t Already On-Chain
+Bucket-related logic will live in a separate `bucketOperations.ts` file. To implement it, take the following steps:
 
-Now that you have the bucket ID, you can ensure the bucket doesn't exist on-chain yet. 
+1. Create a new folder called `operations` within the `src` folder (at the same level as the `services` folder) like so:
 
-```ts title="// **PLACEHOLDER FOR STEP 4: CHECK IF BUCKET EXISTS**"
---8<-- 'code/store-and-retrieve-data/use-storagehub-sdk/create-a-bucket/create-a-bucket.ts:check-bucket'
-```
+    ```bash
+    mkdir operations
+    ```
 
-!!! note
-    The `mspClient` can also be used to check if the derived bucket ID already exists using the `mspClient.buckets.getBucket()` function; however, this only checks if that specific MSP contains a bucket with that ID.
+2. Create a new file within the `operations` folder called `bucketOperations.ts`
 
-If you rerun the script, the response should include:
+3. Add the following code:
 
---8<-- 'code/store-and-retrieve-data/use-storagehub-sdk/create-a-bucket/output-03.html'
+    !!! note
+        Within the `createBucket` helper method, the `getValueProps` helper method from the previous section is used.
 
-## Get Parameters from the MSP
 
-To prepare all the parameters needed for the `createBucket` function, additional data from the MSP is required, such as `mspId` and `valuePropId`.
+    ```ts title="bucketOperations.ts"
+    --8<-- 'code/store-and-retrieve-data/use-storagehub-sdk/create-a-bucket/bucketOperations.ts:imports'
 
-```ts title="// **PLACEHOLDER FOR STEP 5: GET MSP PARAMS**"
---8<-- 'code/store-and-retrieve-data/use-storagehub-sdk/create-a-bucket/create-a-bucket.ts:get-msp-params'
-```
+    --8<-- 'code/store-and-retrieve-data/use-storagehub-sdk/create-a-bucket/bucketOperations.ts:create-bucket'
+    ```
 
-## Create a Bucket
+    With the `createBucket` helper method you go through the full lifecycle of a bucket-creation transaction. First, the `mspId` is fetched and one of the MSP's value props is selected, since a value prop (a file storing payment plan) is one of the required params while creating a bucket. Then, a deterministic bucket ID is derived from your wallet address and chosen bucket name, which ensures the same inputs always generate the same `bucketId`. Before any on-chain transactions, there is a check whether the bucket already exists in the `providers.buckets` mapping to prevent accidental overwrites and to avoid a failing transaction. 
```

Comment:
Breaking this up into a list could make this more readable:

```markdown
    The `createBucket` helper handles the full lifecycle of a bucket-creation transaction:

    - It fetches the MSP ID and selects a value prop (required to create a bucket).
    - It derives a deterministic bucket ID from your wallet address and chosen bucket name.
    - Before sending any on-chain transaction, it checks whether the bucket already exists to prevent accidental overwrites.

    Once the check passes, the `createBucket` extrinsic is called via the StorageHub client, returning the `bucketId` and `txReceipt`.
```